skbio/io/format/gff3.py

```diff
@@ -0,0 +1,469 @@
+semicolons. Multiple values of the same tag are indicated by
+separating the values with the comma ",". The following tags have
+predefined meanings:
+
+* ID. Indicates the unique identifier of the feature. IDs must be
+  unique within the scope of the GFF file.
+
+* Name. Display name for the feature. This is the name to be displayed
+  to the user. Unlike IDs, there is no requirement that the Name be
+  unique within the file.
+
+* Alias. A secondary name for the feature. It is suggested that this
+  tag be used whenever a secondary identifier for the feature is
+  needed, such as locus names and accession numbers. Unlike ID, there
+  is no requirement that Alias be unique within the file.
+
+* Parent. Indicates the parent of the feature. A parent ID can be used
+  to group exons into transcripts, transcripts into genes and so
+  forth. A feature may have multiple parents. Parent can *only* be
+  used to indicate a partof relationship.
+
+* Target. Indicates the target of a nucleotide-to-nucleotide or
+  protein-to-nucleotide alignment. The format of the value is
+  "target_id start end [strand]", where strand is optional and may be
+  "+" or "-". If the target_id contains spaces, they must be escaped
+  as hex escape %20.
+
+* Gap. The alignment of the feature to the target if the two are not
+  collinear (e.g. contain gaps). The alignment format is taken from
+  the CIGAR format described in the Exonerate documentation.
+
+* Derives_from. Used to disambiguate the relationship between one
+  feature and another when the relationship is a temporal one rather
+  than a purely structural "part of" one. This is needed for
+  polycistronic genes.
+
+* Note. A free text note.
+
+* Dbxref. A database cross reference. See the GFF3 specification for
+  more information.
+
+* Ontology_term. A cross reference to an ontology term.
+
+* Is_circular. A flag to indicate whether a feature is circular.
+
+The columns and attributes are read in as the vocabulary defined in
+genbank parsers (:mod:`skbio.io.format.genbank`).
+
+Format Parameters
+-----------------
+
+Reader-specific Parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+``interval_metadata_dict`` is a ``dict`` parameter required for
+``IntervalMetadata`` generator. Its key is the ``str`` of sequence ID
+and its value is the ``IntervalMetadata`` object for the sequence. The
+generator will read the GFF3 file and add the annotation of sequences
+to the corresponding ``IntervalMetadata`` objects. If the annotations of
+some sequences in the GFF3 doesn't match any sequence IDs in the input
+dict of ``interval_metadata_dict``, those annotations will be skipped
+and not parsed.
+
+``interval_metadata`` is an ``IntervalMetadata`` object required for
+``IntervalMetadata`` GFF3 reader. It reads the GFF3 file and add the
+annotations to the input ``interval_metadata``.
+
+``rec_num`` is an ``int` parameter used with the ``IntervalMetadata``,
+``DNA``, and ``Sequence`` GFF3 readers. It specifies which GFF3 record
```

Comment:
Thanks for the details. Based on that I think it makes sense to require passing a `seq_id` when reading a single sequence's annotations. This will avoid odd cases where annotations from different sequences are interleaved, and it isn't clear what annotations you'll get from the "first" sequence/record. Requiring a `seq_id` on read also nicely mirrors the required `seq_id` when writing. Does that sound good to you?